src/algebra/monoid_algebra.lean

```diff
@@ -632,6 +698,45 @@ f.single_mul_apply_aux r _ _ _ $ λ a, by rw [zero_add]
 
 end misc_theorems
 
+end add_monoid_algebra
+
+/-!
+#### Conversions between `add_monoid_algebra` and `monoid_algebra`
+
+While we were not able to define `add_monoid_algebra k G = monoid_algebra k (multiplicative G)` due
+to definitional inconveniences, we can still show the types are isomorphic.
+-/
+
+/-- The equivalence between `add_monoid_algebra` and `monoid_algebra` in terms of `multiplicative` -/
+protected def add_monoid_algebra.to_multiplicative [semiring k] [add_monoid G] :
+  add_monoid_algebra k G ≃+* monoid_algebra k (multiplicative G) :=
+{ map_mul' := λ x y, by {
+    simp only [finsupp.dom_congr],
+    apply add_monoid_algebra.map_domain_mul',
```

Comment:
Note that this line is the only reason `map_domain_mul'` exists - I can't work out any way to invoke the bundled version here.